lib/autoload/sfAutoload.class.php

```diff
@@ -188,7 +188,7 @@ public function loadClass($class)
     $class = strtolower($class);
 
     // class already exists
-    if (class_exists($class, false) || interface_exists($class, false))
+    if (class_exists($class, false) || interface_exists($class, false) || trait_exists($class, false))
```

Comment:
Oops.

We can check that with `function_exists()` or `PHP_VERSION_ID`, but that would mean the framework will behave differently on PHP 5.3 and 5.4+

Not checking trait has been already loaded while supporting autoloading them is inconsistent as well IMO, but I don't think there can be a workaround to make it work on PHP 5.3.

The only thing I can suggest is dropping support for PHP 5.3, which is 8 years old now and has been unsopported for over 3 years:

http://php.net/supported-versions.php